sdks/java/io/mqtt/src/test/java/org/apache/beam/sdk/io/mqtt/MqttIOTest.java

```diff
@@ -193,6 +195,27 @@ public void run() {
     publisherThread.join();
   }
 
+  /**
+   * Test for BEAM-3282: this test should not timeout, and start/advance should return false.
+   */
+  @Test(timeout = 5000L)
+  public void testReceiveWithTimeout() throws Exception {
```

Comment:
I think you could also test this simpler: use withMaxReadTime on a test with TestPipeline/PAssert. withMaxReadTime effectively didn't work before this PR - if you ran out of records before max read time elapses, the pipeline would hang.

Reply:
Good idea ! Let me update that way.